### PR TITLE
weather api added

### PR DESCRIPTION
**Pull request by kumsomi**
This PR consists of weather api and get the data based on the location

### CODE REVIEW

File: src/Components/Weather.jsx [hunk 1]
```diff
@@ -1,8 +1,79 @@
...
+        navigator.geolocation.getCurrentPosition(success, error);
+      };
+    
+      async function getWeather(latitude, longitude) {
+        let apiReq = "";
+        latitude && longitude
+        ? (apiReq = `${url}?key=${key}&q=${latitude},${longitude}&aqi=yes`)
+        : (apiReq = `${url}?key=${key}&q=Bangalore&aqi=yes`);
+        try {
+          const res = await axios.get(apiReq);
```
Comment (anupkgurung) [inline src/Components/Weather.jsx:42]: res can have a better name.
you could destructure the response a.k.a res .

File: src/Components/Weather.jsx [hunk 2]
```diff
@@ -1,8 +1,79 @@
...
+            location: {
+              name: res.data.location.name,
+              region: res.data.location.region,
+              country: res.data.location.country,
+            },
+            temperature: res.data.current.temp_c,
+            icon: res.data.current.condition.icon,
+          });
+        } catch (error) {
+          console.log(error);
```
Comment (anupkgurung) [inline src/Components/Weather.jsx:53]: you can use toast to show error and remove console logs

**Review by anupkgurung — APPROVED** [review]
left few suggestion.
comments can be removed if not needed.

**Review by jeetbhalani15 — APPROVED** [review]

File: src/Components/Weather.jsx (hunk 2 above)
Comment (kumsomi) [reply src/Components/Weather.jsx:53]: Oka, I'll change it. Thanks for the suggestion

File: src/Components/Weather.jsx (hunk 1 above)
Comment (kumsomi) [reply src/Components/Weather.jsx:42]: Okay

**Pull request merged**